extensions/libraries/redcore/form/fields/rdatepicker.php

```diff
@@ -77,13 +230,95 @@ protected function getInput()
 	}
 
 	/**
+	 * Matches each symbol of PHP date format standard
+	 * with jQuery equivalent codeword
+	 *
+	 * @param   string  $jQueryFormat  $jQueryFormat date format
+	 *
+	 * @author Tristan Jahier
+	 *
+	 * @return  string
+	 */
+	protected function dateFormatJQueryUIToPHP($jQueryFormat)
+	{
+		$phpFormat = "";
+		$escaping = false;
+		$length = strlen($jQueryFormat);
+
+		for ($i = 0; $i < $length; $i++)
+		{
+			$char = $jQueryFormat[$i];
+
+			// JQuery date format escaping character
+			if ($char === '\'')
+			{
+				if ($escaping)
+				{
+					$escaping = false;
+				}
+				else
+				{
+					$i++;
+					$phpFormat .= '\\' . $jQueryFormat[$i];
+					$escaping = true;
+				}
+			}
+			else
+			{
+				if ($escaping || !($findMatch = $this->findMatch(substr($jQueryFormat, $i))))
+				{
+					if ($char == ' ')
+					{
+						$phpFormat .= $char;
+					}
+					else
+					{
+						$phpFormat .= "\\" . $char;
+					}
+				}
+				else
+				{
+					$phpFormat .= $this->symbolsMatching[$findMatch];
+					$i += strlen($findMatch) - 1;
+				}
+			}
+		}
+
+		return $phpFormat;
+	}
+
+	/**
+	 * Find match date template
+	 *
+	 * @param   string  $string  Current string
+	 *
+	 * @return bool|string
+	 */
+	protected function findMatch($string)
+	{
+		$length = 0;
+		$itemKey = false;
+
+		foreach ($this->symbolsMatching as $key => $item)
+		{
+			if (strpos($string, $key) === 0 && $length < strlen($item))
+			{
+				$length = strlen($item);
+				$itemKey = $key;
+			}
+		}
+
+		return $itemKey;
+	}
+
+	/**
 	 * Build the datepicker JS options
 	 *
 	 * @return  string  Options in json format
 	 */
 	protected function getDatepickerOptions()
 	{
-		$options = new JRegistry;
+		$return = '{';
```

Comment:
Agree, I changed it, because it converted JSON string values always with quotes, but sometimes we can send anonymous function or regular function like value and quotes, in this case, don't need. This is the regular situation for javascript - use JSON construction, where value can be a function. But PHP json_encode can not allow it.